Add SSOMetadataURL example (#145)
Mirantis/mke-docs: content/docs/tutorials/authentication-provider-setup/setting-up-okta-as-a-saml-provider.md

</ds:X509Data>
</ds:KeyInfo>
</md:KeyDescriptor>
<md:NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified</md:NameIDFormat>
<md:NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress</md:NameIDFormat>
<md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://dev-64105006.okta.com/app/dev-63105106_mke_2/exk75pi5do2MzU1t95r7/sso/saml"/>
<md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="https://dev-63105106.okta.com/app/dev-63105106_mke_2/exk75pi5do2MzU1t95r7/sso/saml"/>
</md:IDPSSODescriptor>
</md:EntityDescriptor>

> ```

Author: Carry over the discussion from the previous PR

This is useful for a customer when they are setting up SAML and they wonder what they should be seeing. If the goal of the docs is to "help customers setup an MKE4 cluster", then I suggest we leave this and look for more areas that we can clarify what should be seen for customers.

{{< /tab >}}

{{< tab >}}

Manually configuring the SAML service in MKE allows more control over the configuration.

> 
> ```yaml

Reviewer: ```suggestion
An example of the `saml` section of the MKE 4 configuration file under `authentication` follows:

```yaml
```

24x+dmDWEimHSeXrvKSzObI4WzxP2vAmsO+apmNW60ECAwEAATANBgkqhkiG9w0BAQsFAAOCAQEA
qUBeR/TR7tqS3MH8uY+JnzzxJKAhb9bWnMgDM3msIyLAF8CuRSC283Ws5+t11QF+C4pEV4zXRjrM
Z1mNWj4eQEwB4+ZdTy4JhpBavOZSS2WESdAsnRwrYOb+t3FBae7Ay+lPnos5UPjgpV1aw3A6/BYe
kDBBZOSieSGOa6Me2Qx358bb2oiFqBGgXoI22rgmDBvrxTsNMv/4T3/2i3cOjF6YC1vdYLlmqPxw
2HK2OIU6yrLsvYz0W3VZwJFzMAmEPYAwmdzT1G/qpzcuVXJUIy0TS7GuBghcQ8Zfy3ya18aUBlwb
mcw9Dlufaan1XxqnVivuWe3qVSVIWngUOUt2EA==
-----END CERTIFICATE-----
```

This information can be found in Okta b y navigating to

1. Applications -> Applications -> Your application
2. Click the **Sign On** tab
3. Click the `View SAML setup instructions` link on the right side of the page.
4. The `ssoURL` is displayed in the `Identity Provider Single Sign-On URL` section.
5. The certificate is displayed in the `X.509 Certificate` section.
6. The `redirectURI` will be the external address of your cluster with `/dex/callback` appended to the end.
7. The `usernameAttr` and `emailAttr` are the attributes that will be used to map the user's name and email to the MKE user. These can be left the same as the values in the example since that is what Okta was configured to use above.

{{< /tab >}}

{{< /tabs >}}

> Run the `mkectl apply` command with your MKE configuration file.

Reviewer: ```suggestion
Once the `saml` section of the MKE 4 configuration file is set, run it with the `mkectl apply` command.
```

10. Download the certificate to the system from which you will run mkectl:
    a. Navigate to the SAML **Signing Certificates** section.
    b. Click **Actions** for the active certificate and initiate the download.

## Configure MKE

{{< tabs items="SSO metadata URL,manual" >}}

{{< tab >}}
Okta provides a metadata URL with which you can configure SAML for MKE 4. MKE is able to obtain information for all SAML configurations in your MKE cluster through that URL, which you configure to the `ssoMetadataURL` parameter in the `authentication` section of the MKE configuration file.

```yaml
authentication:
  saml:
    enabled: true
    ssoMetadataURL: https://dev-14305804.okta.com/app/exkdtjvsinbvwD5ms5d0/sso/saml/metadata
```

To obtain the metadata URL in Okta, navigate to **Applications -> Applications -> Your application** and click the **Sign On** tab.

> Example ssoMetadataURL information:

Reviewer: ```suggestion
Example `ssoMetadataURL` information:
```

> Example of information provided when you access the `ssoMetadataURL` URL:

Author: ```suggestion
Example ssoMetadataURL information:
```

</ds:X509Data>
</ds:KeyInfo>
</md:KeyDescriptor>
<md:NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified</md:NameIDFormat>
<md:NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress</md:NameIDFormat>
<md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://dev-64105006.okta.com/app/dev-63105106_mke_2/exk75pi5do2MzU1t95r7/sso/saml"/>
<md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="https://dev-63105106.okta.com/app/dev-63105106_mke_2/exk75pi5do2MzU1t95r7/sso/saml"/>
</md:IDPSSODescriptor>
</md:EntityDescriptor>
```

{{< /tab >}}

{{< tab >}}

Set up the SAML service manually in MKE to gain more control over the configuration.

> An example of the `saml` section of the MKE 4 configuration file under `authentication` follows:

Reviewer: ```suggestion

An example of the `saml` section of the MKE 4 configuration file under `authentication` follows:
```

10. Download the certificate to the system from which you will run mkectl:
    a. Navigate to the SAML **Signing Certificates** section.
    b. Click **Actions** for the active certificate and initiate the download.

## Configure MKE

{{< tabs items="SSO metadata URL,manual" >}}

{{< tab >}}

> Okta provides a metadata URL with which you can configure SAML for MKE 4. MKE is able to obtain information for all SAML configurations in your MKE cluster through that URL, which you configure to the `ssoMetadataURL` parameter in the `authentication` section of the MKE configuration file.

Reviewer 2: ```suggestion
Okta provides a metadata URL with which you can configure SAML for MKE 4.
MKE is able to obtain information for all SAML configurations in your MKE cluster through that URL,
which you configure to the `ssoMetadataURL` parameter in the `authentication` section of the MKE
configuration file.
```
Broken between few lines to meet our style guidelines.

|:--------------:|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|  `ssoURL`      | Obtain the `ssoURL` and certificate information from Okta:<br>  1. Navigate to **Applications -> Applications -> Your application**.<br>  2. Click the **Sign On** tab.<br>  3. Click the **View SAML setup instructions` link**. The `ssoURL` displays in the `Identity Provider Single Sign-On URL` section. |
| `redirectURI`  | The external address of your cluster, with `/dex/callback` appended.                                                                                                                                                                                                                                           |

> | `usernameAttr` | Attribute used to map the username to the MKE user.                                                                                                                                                                                                                                                            |

Reviewer 2: ```suggestion
| `usernameAttr` | Attribute that maps the username to the MKE user.                                                                                                                                                                                                                                                            |
```
Changed to match the attribute description below.
